src/reducers/ui.js

```diff
@@ -30,6 +32,10 @@ function update(state = State(), action: Action): Record<UIState> {
     case constants.SHOW_SOURCE: {
       return state.set("shownSource", action.sourceUrl);
     }
+
+    case constants.HIGHLIGHT_SOURCE: {
```

Comment:
Is it possible that we can reuse the `SHOW_SOURCE` action with an additional property? This case is so similar to it.